.github/actions/run-smoke-tests/action.yml

```diff
@@ -0,0 +1,66 @@
+name: Run Smoke Tests
+description: Executes Smoke Tests
+inputs:
+  role-to-assume:
+    description: an IAM role to use in tests
+    required: true
+  region:
+    description: an AWS region to run in
+    required: false
+    default: us-west-2
+runs:
+  using: 'composite'
+  steps:
+    - name: Verify Amplify Path
+      shell: bash
+      run: |
+        if [[ -z "$AMPLIFY_PATH" ]]; then
+          echo "AMPLIFY_PATH must be defined"
+          exit 1
+        fi
+        echo "AMPLIFY_PATH=$AMPLIFY_PATH"
+        $AMPLIFY_PATH version
+
+    - name: Install Dependencies
+      shell: bash
+      run: yarn install
+
+    - name: Build
+      shell: bash
+      run: yarn build-tests
+
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@04b98b3f9e85f563fb061be8751a0352327246b0 # v3.0.1
+      with:
+        role-to-assume: ${{ inputs.role-to-assume }}
+        aws-region: ${{ inputs.region }}
+
+      # For iOS tests
+    - name: Setup Ruby
+      uses: ruby/setup-ruby@250fcd6a742febb1123a77a841497ccaa8b9e939 # v1.152.0
+      if: runner.os == 'macOS'
+      with:
+        ruby-version: '3.2.1'
+
+      # For iOS tests
+    - name: Set Default Xcode Version
+      if: runner.os == 'macOS'
+      shell: bash
+      run: |
+        sudo xcode-select -s "/Applications/Xcode_14.1.app"
+        xcodebuild -version
+
+    - name: Run Smoke Tests
+      shell: bash
+      run: yarn smoketest -- --forceExit --no-cache --maxWorkers=2
+      env:
+        CLI_REGION: ${{ inputs.region }}
+        CI: true
+        CIRCLECI: true
+
+    - name: Upload Report
+      uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce #v3.1.2
+      if: always()
```

Comment:
is this not the default?

Reply:
No. By default step is skipped if previous step fails. This makes sure that report is uploaded regardless of test success.

Reply:
see https://stackoverflow.com/questions/58858429/how-to-run-a-github-actions-step-even-if-the-previous-step-fails-while-still-f